core/state_transition.go

```diff
@@ -512,15 +512,13 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 		vmerr error // vm errors do not effect consensus and are therefore not assigned to err
 	)
 	if contractCreation {
-		ret, _, st.gasRemaining, vmerr = st.evm.Create(sender, msg.Data, st.gasRemaining, value)
+		ret, _, st.gasRemaining, vmerr = st.evm.Create(sender, msg.Data, st.gasRemaining, value, rules.IsPrague)
 		// Special case for EOF, if the initcode or deployed code is
 		// invalid, the tx is considered valid (so update nonce), but
 		// is to be treated as an exceptional abort (so burn all gas).
 		if errors.Is(vmerr, vm.ErrInvalidEOFInitcode) {
-			st.gasRemaining = 0
```

Comment:
Comment still says "is to be treated as an exceptional abort (so burn all gas).". We need to update that (note: I don't really mean to pile more work on @shemnon , just noting it).